buildstock_query/aggregate_query.py

```diff
@@ -305,13 +435,230 @@ def get_lower_timestamps(day, hour):
+            rows_per_building = self._bsq._get_rows_per_building()
+            grouping_metrics_selection = [
+                (safunc.sum(1) / rows_per_building).label("sample_count"),
+                safunc.sum(total_weight / rows_per_building).label("units_count"),
+            ]
+        elif params.timestamp_grouping_func:
+            colname = self._bsq.timestamp_column_name
+            bldg_id_col = bs_tbl.c[self._bsq.building_id_column_name]
+            grouping_metrics_selection = [
+                safunc.count(sa.func.distinct(bldg_id_col)).label("sample_count"),
+                (
+                    safunc.count(sa.func.distinct(bldg_id_col))
+                    * safunc.sum(total_weight)
+                    / safunc.sum(1)
+                ).label("units_count"),
+                (safunc.sum(1) / safunc.count(sa.func.distinct(bldg_id_col))).label("rows_per_sample"),
+            ]
+            sim_info = self._bsq._get_simulation_info()
+            time_col = bs_tbl.c[self._bsq.timestamp_column_name]
+            if sim_info.offset > 0:
+                # If timestamps are not period beginning we should make them so for timestamp_grouping_func aggregation.
+                new_col = sa.func.date_trunc(
+                    params.timestamp_grouping_func, sa.func.date_add(sim_info.unit, -sim_info.offset, time_col)
+                ).label(colname)
+            else:
+                new_col = sa.func.date_trunc(params.timestamp_grouping_func, time_col).label(colname)
+
+            # If include_savings is True, then the order of the columns is different. Do this
+            # to match the behavior of savings_shape query. Can be simplified after savings_shape function is removed.
+            if params.include_savings:
+                group_by_selection.append(new_col)
+            else:
+                group_by_selection.insert(time_indx, new_col)
+        else:
+            time_col = bs_tbl.c[self._bsq.timestamp_column_name].label(self._bsq.timestamp_column_name)
+            grouping_metrics_selection = [
+                safunc.sum(1).label("sample_count"),
+                # '1 *' included in savings query to match existing behavior for testing.
+                # Can be removed after saving_shape is removed.
+                safunc.sum(1 * total_weight if params.include_savings else total_weight).label("units_count"),
+            ]
+            if params.include_savings:
+                group_by_selection.append(time_col)
+            else:
+                group_by_selection.insert(time_indx, time_col)
+
+        # If include_savings is True, then the order of the columns is different. Do this
+        # to match the behavior of savings_shape query. Can be simplified after savings_shape function is removed.
+        if params.include_savings:
+            if params.annual_only or params.timestamp_grouping_func == "year":
+                query_cols = grouping_metrics_selection + query_cols + group_by_selection
```

Comment:
overriding variables (e.g., query_cols) make it hard to read the code and do debugging later